core/src/macosX64Test/kotlin/AdvertisementTest.kt

```diff
@@ -0,0 +1,59 @@
+package com.juul.kable.test
+
+import com.juul.kable.Advertisement
+import com.juul.kable.toNSData
+import platform.CoreBluetooth.CBAdvertisementDataManufacturerDataKey
+import platform.CoreBluetooth.CBPeripheral
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+class AdvertisementTest {
+    @Test
+    fun manufacturerDataCanContainData() {
```

Comment:
For ease of understandability, I like to give my test function names a particular format:

fun [name of item being tested]_[input conditions]_[expected results]

I think in this case, it would be something like:

fun manufacturerData_validAdvertisement_returnsValidData()

Not saying that we need to use that exact format, but doing something akin to that makes it really easy to quickly understand what is being tested, under what conditions, and what the expected results are--a very useful thing to understand should the test ever fail.

Reply:
I kind of go back and forth on what the best format for test function names are.

> `[name of item being tested][input conditions][expected results]`

I used to subscribe the this format as well, but found it too restrictive at times.

I'm happy to keep this flexible (and let the developer decide what the best way to convey what the test is doing). As seeing that `manufacturerDataCanContainData` failed, seems reasonable enough to me to know what went wrong.

Happy if it is changed as well though. (I personally am fine with either naming scheme)

Reply:
Well, what I find useful about using the "_", is that you're able to easily parse out all the components of the test (system under test, input conditions, expected results).

Camelcasing this info together will still contain these components, but it's more difficult to determine what they are, and could be interpreted different ways. 

In what way have you found the "_" format restrictive?

Reply:
> In what way have you found the "_" format restrictive?

Some very small/simplified tests don't adhere to the structure:

1. name of item being tested
2. input conditions
3. expected results

Reply:
As these are the first tests in this codebase I'll go ahead and switch to the `fun [name of item being tested]_[input conditions]_[expected results]` naming scheme to set precedent.

And thus defer further argument to our future selves. 🤠

Reply:
> > In what way have you found the "_" format restrictive?
> 
> Some very small/simplified tests don't adhere to the structure:
> 
> 1. name of item being tested
> 2. input conditions
> 3. expected results

That's true. This very example could fit that case. Like, in, "manufacturerData_validAdvertisement_returnsValidData()", "validAdvertisement" might be superfluous (if there is a use case where there could be an invalid advertisement, I'd say it isn't). In that scenario, I think it'd be fine to shorten to `fun manufacturerData_returnsValidData()` as this still clearly denotes what is being tested and the outcome--with the input being understood as "happy path default conditions."

Reply:
Good example. Ultimately, I'm pretty flexible on the naming of test functions. 😄